antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -16,7 +25,189 @@
+                ],
+            )
+        elif existing_operator == BindingConstraintOperator.LESS:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_target_path(is_link_lt)],
+            )
+        elif existing_operator == BindingConstraintOperator.GREATER:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_target_path(is_link_gt), matrix_node_gt.infer_path()],
+            )
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    elif existing_operator == BindingConstraintOperator.BOTH:
+        if new_operator == BindingConstraintOperator.EQUAL:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_eq.infer_target_path(is_link_lt)],
+            )
+        elif new_operator == BindingConstraintOperator.LESS:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_lt.infer_target_path(is_link_lt)],
+            )
+        elif new_operator == BindingConstraintOperator.GREATER:
+            return MatrixInputTargetPaths(
+                matrix_input_paths=[matrix_node_lt.infer_path(), matrix_node_gt.infer_path()],
+                matrix_target_paths=[matrix_node_gt.infer_target_path(is_link_gt)],
+            )
+        else:
+            raise NotImplementedError(
+                f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+            )
+    else:
+        raise NotImplementedError(
+            f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+        )
+
+
+def _update_matrices_names(
+    file_study: FileStudy,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> None:
+    """
+    Update the matrix file name according to the new operator.
+
+    Args:
+        file_study: the file study
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Raises:
+        NotImplementedError: if the case is not handled
+    """
+
+    if existing_operator == new_operator:
+        return
+
+    parent_folder_node = file_study.tree.get_node(["input", "bindingconstraints"])
+    assert isinstance(parent_folder_node, FolderNode), f"Node type not handled yet: {type(parent_folder_node)}"
```

Comment:
Remove this check it doesn't make sense

Reply:
you need to assert the type for the function that comes just after `_infer_input_and_target_paths` 
otherwise `mypy` would not not be ok with it.